util/tls/tls.go

```diff
@@ -0,0 +1,68 @@
+package tls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"os"
+)
+
+type config struct {
+	CaFile   string `yaml:"ca_file"`
+	CertFile string `yaml:"cert_file"`
+	KeyFile  string `yaml:"key_file"`
+}
+
+type Tls struct {
+	config
+	tlsConfig *tls.Config
+}
+
+func (tlsConfig *Tls) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var config config
+	err := unmarshal(&tlsConfig.config)
+	if err != nil {
+		return err
+	}
+
+	if tlsConfig.config.CaFile == "" ||
```

Comment:
If some are specified and not others should we err?

Reply:
Done.